File: 2581-divide-players-into-teams-of-equal-skill/2581-divide-players-into-teams-of-equal-skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        int left = 0; 
        int right = skill.size() - 1;
        long long chem = 0;  
        sort(skill.begin(), skill.end()); 

        int Sum = skill[right] + skill[left]; 

        while(left < right){
            if(skill[right] + skill[left] != Sum) return -1; 
            chem += (skill[right] * skill[left]);

            left++;
            right--; 
        }

        return chem; 


    }
};

File: 2581-divide-players-into-teams-of-equal-skill/2581-divide-players-into-teams-of-equal-skill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        skill.sort()

        left, right, ans = 0, len(skill) -1 , 0

        startingSum = skill[left] + skill[right]

        while left < right: 
            prod = skill[left] * skill[right]
            if(skill[left] + skill[right] == startingSum):
                ans += prod
            else: 
                return -1
            left += 1
            right -= 1
        
        return ans 